src/components/person.tsx

```diff
@@ -0,0 +1,91 @@
+import classNames from 'classnames';
+import { useContext } from 'react';
+import { Link, useLocation } from 'react-router-dom';
+import { MyContext } from './state';
+import { Person as PersonType } from '../types';
+
+interface PersonComponentProps {
+  pers: PersonType;
+}
+
+export const Person: React.FC<PersonComponentProps> = ({ pers }) => {
+  const { name, sex, born, fatherName, motherName, slug, died } = pers;
+  const { setSelectedPerson, person } = useContext(MyContext);
+  const location = useLocation();
+
+  const motherPerson = person.find(
+    (perso: PersonType) => perso.name === motherName,
+  );
+  const fatherPerson = person.find(
+    (perso: PersonType) => perso.name === fatherName,
+  );
+
+  const isPersonInUrl = location.pathname.includes(slug);
+
+  const getPeopleClass = () =>
+    classNames({ 'has-background-warning': isPersonInUrl });
+
+  return (
+    <tr data-cy="person" key={name} className={getPeopleClass()}>
+      <td>
+        <Link
+          to={`/people/${slug}`}
+          className={classNames({
+            'has-text-danger': sex === 'f',
+            'has-background-warning': isPersonInUrl,
+          })}
+          onClick={() => setSelectedPerson(slug)}
+        >
+          {name}
+        </Link>
+      </td>
+
+      <td>{sex}</td>
+      <td>{born}</td>
+      <td>{died}</td>
+      <td>
+        {motherName ? (
+          motherPerson ? (
+            <Link
+              to={`/people/${motherPerson.slug}`}
+              className={classNames({
+                'has-text-danger': true,
+                'has-background-warning': location.pathname.includes(
+                  motherPerson.slug,
+                ),
```

Comment:
The repeated use of `location.pathname.includes` could be optimized by storing the result in a variable outside of the JSX to avoid recalculating it multiple times for each render. Consider computing this once at the top of the component and reusing the value.